### DISCUSSION TITLE
Storage solutions

### DISCUSSION THREAD

**Discussion by stygigoth** [modrinth-smp/pack#24] [General]
Currently our only options for storage are chests, barrels, or item vaults. Chests & barrels are easy to use, but have immense performance issues. Item vaults are actually implemented fairly well, but are hard to interface with. I propose adding a mod like [excorrelation](https://modrinth.com/mod/excorrelation) to address these issues.

Personally, I've noticed massive amounts of lag near large amounts of chests, and if there are enough, it makes the game nearly unplayable.

**Comment by ghost** [comment]
Can't you put backpacks into chests?

**Comment by stygigoth** [comment]
You can, although that has the same limitation as item vaults, just to a lesser degree. Trying to dig through a chest full of backpacks to find the right one sounds like hell, not to mention the resources that go into making them.

**Comment by ghost** [comment]
A lot of thought needs to be put into adding mods, especially storage mods. You can't reamove a content mod without casuing damage. We need to be sure we 1000% need this.

**Comment by Gaming32** [comment]
I personally suggest [EZ Storage 2](https://www.curseforge.com/minecraft/mc-mods/ezstorage-2), but ported to 1.18.2.

**Comment by stygigoth** [comment]
I see the appeal of a simple mod like EZ Storage 2, but looking at the repos of both mods a bit it seems that in addition to a port, there would likely be a few bugs cropping up that would need fixed. Either that, or it's a rewrite and not a port, which, I imagine, would take quite some time. I'm not saying it isn't worth it, but this needs considered too.

**Comment by Gaming32** [comment]
When I say port, I generally mean rewrite. Basically create a new project in the newer version, and translate code line-by-line. That would kinda be the necessary approach, as it would be porting across both versions (including The Flattening) and loaders. I kinda wanna do it though, it seems interesting to me, as I've never done it before.

**Comment by Gaming32** [comment]
I have now finished porting EZStorage 2, for the most part. It's in the pack now.
